From 431650453771ec231b0a556579055fdfea04d7bf Mon Sep 17 00:00:00 2001
From: Steev Klimaszewski <steev@kali.org>
Date: Thu, 16 Feb 2023 14:44:38 -0600
Subject: [PATCH 201/230] hack: Set bid to b if 0

The bluetooth driver seems to get the wrong bid[0] for the Thinkpad
X13s, so if bid[0], set it to 0x0b so it correctly sets it to bXX for
the nvm firmware.

Additionally, get rid of the useless prints, since most users don't need
them.

Not-signed-off-by: Steev Klimaszewski <steev@kali.org>
---
 drivers/bluetooth/btqca.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/bluetooth/btqca.c b/drivers/bluetooth/btqca.c
index 67dd99e9c27..570722a7858 100644
--- a/drivers/bluetooth/btqca.c
+++ b/drivers/bluetooth/btqca.c
@@ -630,7 +630,6 @@ static void qca_generate_nvm_name(struct hci_dev *hdev, char *fwname,
 
 	soc_ver = get_soc_ver(ver.soc_id, ver.rom_ver);
 	rom_ver = ((soc_ver & 0x00000f00) >> 0x04) | (soc_ver & 0x0000000f);
-	bt_dev_info(hdev, "bid == 0x%02x 0x%02x", bid[0], bid[1]);
 
 	if ((ver.soc_id & 0x0000ff00) == 0x1200) /*hsp gf chip*/
 		variant = "g";
@@ -640,10 +639,10 @@ static void qca_generate_nvm_name(struct hci_dev *hdev, char *fwname,
 	if (bid[0] == 0 && bid[1] == 0) {
 		snprintf(fwname, max_size, "qca/hpnv%02x%s.bin", rom_ver, variant);
 	} else {
+		/* Horrible hack to override boardid = 0 to be b. */
 		snprintf(fwname, max_size, "qca/hpnv%02x%s.%x%02x",
-			rom_ver, variant, bid[0], bid[1]);
+			rom_ver, variant, bid[0] == 0 ? 0x0b : bid[0], bid[1]);
 	}
-	bt_dev_info(hdev, "%s:  %s", __func__, fwname);
 }
 
 int qca_uart_setup(struct hci_dev *hdev, uint8_t baudrate,
-- 
2.39.0

